### DISCUSSION TITLE
Set multiple keyhandlers at once or have configuration file?

### DISCUSSION THREAD

**Discussion by SteveL-MSFT** [lzybkr/PSReadLine#2382] [Ideas]
A good portion of my `$profile` is to set PSReadLine options and keyhandlers.  I wonder if it makes sense to have `Set-PSReadLineKeyHandler` accept a hashtable of chords and functions/scriptblocks to allow setting multiple keyhandlers with one call to the cmdlet or alternatively, have `Set-PSReadLineConfiguration` and `Export-PSReadLineConfiguration` where `Export` would produce a json (?) file that contains all options and keyhandlers that could then be fed to `Set` to apply.  This might also make it easier to share options/keyhandlers.

**Comment by Nevember** [comment]
Honestly, when I was first reading about PSReadLine and when the readme referenced [the sample profile](https://github.com/PowerShell/PSReadLine/blob/master/PSReadLine/SamplePSReadLineProfile.ps1), I'd hoped it meant that PSReadLine supported loading its own .ps1 profile. While I know now that that's not the case, I was excited about that prospect, since, like yours, much of my $profile is devoted to PSReadLine options.

With that in mind, I really like the idea of being able to export/import PSReadLine options.

**Comment by lzybkr** [comment]
A configuration file was definitely considered, probably even my first thought.

I went with cmdlets for configuration because:
* Familiarity, no new syntax to learn.
* They are more flexible, e.g. to conditionally apply a setting.
* Syntax looks much better when binding scriptblocks - yaml would be horrible.
* There wasn't any widely used configuration file design for PowerShell to adopt.

Binding multple key handlers with one invocation might be handy, but it does feel a bit like yak shaving at this point.
